card/mod/edit/set/all/bridge/tab_visibility.rb

```diff
@@ -0,0 +1,51 @@
+format :html do
+  def visible_bridge_tabs
+    BRIDGE_TABS.select do |key, _title|
+      send "show_#{key}?"
+    end
+  end
+
+  private
+
+  def show_account_tab?
+    return unless card.real?
+
+    card.account && card.ok?(:update)
+  end
+
+  def show_engage_tab?
+    return unless card.real?
+
+    show_follow? || show_discussion?
+  end
+
+  def show_history_tab?
+    card.real?
+  end
+
+  def show_related_tab?
+    card.real?
+  end
+
+  def show_rules_tab?
+    true
```

Comment:
I suspect some sharks are going to want to be able to hide this tab from butterflies/eagles.  But this is a very clean starting place.